ci/docker/install/ubuntu_docs.sh

```diff
@@ -21,8 +21,44 @@
 # the whole docker cache for the image
 
 set -ex
-wget http://downloads.lightbend.com/scala/2.11.8/scala-2.11.8.deb && \
-    dpkg -i scala-2.11.8.deb && rm scala-2.11.8.deb
+# Install dependencies
+echo 'Installing dependencies...'
+apt-get install -y \
+    apt-transport-https \
+    build-essential \
+    ca-certificates \
+    curl \
+    doxygen \
+    git \
+    libatlas-base-dev \
```

Comment:
Are all these dependencies really required? Installing them here might override other version restrictions. It would be better to run this script after the other install scripts and then see what's actually required for the document generation.

Reply:
I'd like to just bake this stuff into a docker image. But to answer the question: these are all of the requirements needed to build the docs. Except R. That takes more deps and isn't included in the required scope for the nightlies. It is possible that you can get *a result* without some of these, but I resolved warning after warning and error after error until I had a *viable* result.

Refer to: https://github.com/apache/incubator-mxnet/issues/10273
See render in staging:
http://107.21.90.138/api/python/ndarray/ndarray.html#mxnet.ndarray.Correlation

and
https://github.com/apache/incubator-mxnet/issues/10226
Fixed in staging:
http://107.21.90.138/versions/master/api/python/io/io.html#mxnet.io.ImageRecordIter

Also, this is the script for the ubuntu_docs, so why does overriding anything matter? It's for docs. It's what the docs need to be happy.

Reply:
Overriding matters because this script is part of the ubuntu_cpu dockerfile pipeline (https://github.com/apache/incubator-mxnet/blob/master/ci/docker/Dockerfile.build.ubuntu_cpu#L45). 

I don't think we need to install libjemalloc-dev, liblapack-dev, libopenblas-dev, libopencv-dev, numpy, git and all the other libraries at this point since they should already be present in the underlying layers (see https://github.com/apache/incubator-mxnet/blob/master/ci/docker/install/ubuntu_core.sh#L27).

Reply:
None of these are versioned at any point. I'd argue that while layering makes sense, it opens things up to breakage when we don't know what the full stack is for docs. For example, one of my deps, `libatlas-base-dev ` is only appearing in the Caffe script. If scripts gets pulled out for some reason, then docs may break. 

`libjemalloc-dev` doesn't appear anywhere in the other scripts despite every build run telling you that you should have it installed (to massive improve speed, or so it contends).

These seem like they should move to the base script:
```
    apt-transport-https \
    ca-certificates \
    software-properties-common \
```

These would seem to not be relevant to CI, but I imagine if tutorials are being automatically executed then they're required, but at what layer? Just for docs?
```
    curl \
    unzip \
    wget
```

Reply:
Right, at the moment they're not versioned. The idea here is to make sure every single dependency is only defined at one place and thus giving us the ability to pin the version if required - without having to look through all the scripts and potentially missing a duplicate. 

Fir libatlas and libjemalloc, feel free to move them to core as they are definitely no api specific dependencies. 

curl, unzip and wget are required by some install scripts and the runtime as well if I'm not mistake. They're also a core dependency.